### PR TITLE
Implement tooling for building artifacts and deploying with artifacts

### PR DESCRIPTION
**Pull request by odyssey4me**
This PR adds the tooling to build apt, git, python, container artifacts (the pipeline to build them is implemented in that order) and adds the configuration changes to consume these artifacts for a RPC-O deployment.

Note:
At this stage the jobs have not been changed appropriately, so the test jobs in rpc-o master will likely all fail with the changed deployment configuration here. This PR is up for review in order to get eyes on the changes made while the jobs are changed in the background.

Connects https://github.com/rcbops/u-suk-dev/issues/1668

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by evrardjp** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_xenial

**Comment by hughsaunders** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_ceph

**Comment by evrardjp** [comment]
I can't vote for it, because I was part of the work, but I think it looks good.

**Comment by odyssey4me** [comment]
recheck_cit_swift

**Comment by odyssey4me** [comment]
recheck_cit_swift failed due to disk space issue on jenkins node

**Comment by odyssey4me** [comment]
recheck_cit_swift failed due to cloud node allocation failure

**Comment by major** [comment]
Wow, that's quite a patch.

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
recheck_cit_xenial rpc-repo was busy being updated

**Comment by odyssey4me** [comment]
recheck_cit_xenial

**Comment by evrardjp** [comment]
@major I think we managed to narrow it down to 14 commits, instead of the 90+ initially. I think it's readable now. Good luck!

**Comment by alextricity25** [comment]
Reviewing this now.

File: rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml [hunk 1]
```diff
@@ -30,3 +30,171 @@ pkg_locations:
   - /etc/openstack_deploy
   - /opt/rpc-openstack/rpcd
 
+# TODO(odyssey4me):
```
Comment (odyssey4me) [inline rpcd/etc/openstack_deploy/user_osa_variables_defaults.yml]: Yes, as the group_vars patch only recently merged I haven't had the chance to switch this config over. I've been focusing on trying to get this passing builds so that we can merge it to unblock the Ocata work. My intent will be to push up another PR on top of this to do that.

**Comment by odyssey4me** [comment]
Closed, then opened the PR to try and trigger a fresh PR check in Travis which is bugging out due to the history rewriting.

**Comment by evrardjp** [comment]
woot!
